ros2/rust/string_length_node/src/main.rs

```diff
@@ -0,0 +1,69 @@
+// Rust imports
+use std::sync::{Arc, Mutex};
+// rclrs related imports
+use rclrs::{Publisher, RclrsError};
+use std_msgs::msg::String as StringMsg;
+use std_msgs::msg::UInt8 as UInt8Msg;
+
+// Define the struct for the node
+struct StringLengthNode {
+    node: Arc<rclrs::Node>,
+    _subscription: Arc<rclrs::Subscription<StringMsg>>,
+    data: Arc<Mutex<Option<StringMsg>>>,
+    publisher: Arc<rclrs::Publisher<UInt8Msg>>,
+}
+
+// Define the implementation of the node.
+impl StringLengthNode {
+    // This function is called when creating the node.
+    fn new(context: &rclrs::Context) -> Result<Self, rclrs::RclrsError> {
+        let node = rclrs::Node::new(context, "string_length_node")?;
+        let data = Arc::new(Mutex::new(None));
+        let data_cb = Arc::clone(&data);
+        let _subscription = node.create_subscription(
+            "string_topic",
+            rclrs::QOS_PROFILE_DEFAULT,
+            move |msg: StringMsg| {
+                *data_cb.lock().unwrap() = Some(msg);
```

Comment:
If you get a second message before you've consumed this one, you'll overwrite it.

You could either buffer the messages to a Mutex<Vec<StringMsg>>, or you could use a channel: https://doc.rust-lang.org/rust-by-example/std_misc/channels.html

Reply:
Added the buffer